### PR TITLE
[with_data_parallel][part7] remove with_data_parallel in custom op test

### PR DESCRIPTION
**Pull request by kangguangli**
### PR types

Others
### PR changes

Others
### Describe

This PR deletes the `with_data_parallel` in custom op related unit tests and add a new one aims to test the custom op when training with multi-device.

#### Others
Pcard-67004

### CODE REVIEW

File: python/paddle/fluid/tests/custom_op/CMakeLists.txt [hunk 1]
```diff
@@ -19,10 +19,21 @@ if(WITH_GPU OR APPLE)
   endif()
 endif()
 
+if(WITH_GPU)
+  py_test(test_custom_op_relu_model_static_multidevice
+          SRCS test_custom_op_relu_model_static_multidevice.py)
+  set_tests_properties(test_custom_relu_model
```
Comment (jiahy0825) [inline python/paddle/fluid/tests/custom_op/CMakeLists.txt]: `test_custom_relu_model `->`test_custom_op_relu_model_static_multidevice`

**Review by jiahy0825 — COMMENTED** [review]
We also need to assert the outputs between `custom_relu` and `relu` are equal.

File: python/paddle/fluid/tests/custom_op/CMakeLists.txt (hunk 1 above)
Comment (kangguangli) [reply python/paddle/fluid/tests/custom_op/CMakeLists.txt]: Done. And Thanks.

**Comment by kangguangli** [comment]
> We also need to assert the outputs between `custom_relu` and `relu` are equal.

Has added checks about relu output.

**Review by jiahy0825 — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM
单测执行时间设置

**Pull request merged**